lms/static/scripts/frontend_apps/components/BasicLtiLaunchApp.js

```diff
@@ -197,17 +195,26 @@ export default function BasicLtiLaunchApp() {
 
     /**
      * Callback when the selected student changes. This function
-     * changes the key on the iframe so it gets rebuilt.
+     * makes an rpc call to change to the focused user of the sidebar.
      */
-    const changeSelectedUserKey = userid => {
-      setSidebarKey(userid);
+    const changeSelectedUser = user => {
+      const sidebar = getSidebarWindow();
+
+      if (sidebar) {
```

Comment:
I'll add a comment about this, but LMSGrader called this right away because of line 43 in LMSGrader

`onChangeSelectedUser({}); // any non-real userid will clear out a previously focused user`

This is necessary to clear out the user, but its also the same code path for when there is no valid student (e.g. All Students). What the means is onChangeSelectedUser is triggered right away because `useEffect` always fire once at the start. 

I think all of that is okay, but where it breaks down is that the sidebar is usually not ready at the outside so we don't kick off an RPC message. It only works because there is no focused user at the start and we don't need to clear out the user since its not there already.

Reply:
An improvement here could be that getSidebarWindow() could be async, what do you think?